src/http/api/v1/matches.ts

```diff
@@ -111,5 +113,61 @@ export const createHandler = (config: HttpApiConfig): express.Router => {
     res.send(responseBody);
   }
 
+  router.post("/chains/:chainId/rounds/:roundId/estimate", async (req, res) => {
+    await estimateMatchesHandler(req, res, 200);
+  });
+
+  async function estimateMatchesHandler(
+    req: Request,
+    res: Response,
+    okStatusCode: number
+  ) {
+    const chainId = Number(req.params.chainId);
+    const roundId = req.params.roundId;
+    const potentialVotes = potentialVotesSchema
+      .parse((req.body as { potentialVotes: PotentialVotes }).potentialVotes)
+      .map((vote) => ({
+        ...vote,
+        amount: BigInt(vote.amount),
+      }));
+
+    const chainConfig = config.chains.find((c) => c.id === chainId);
+    if (chainConfig === undefined) {
+      throw new Error(`Chain ${chainId} not configured`);
```

Comment:
This will return a 500 error, it should return a 4xx error to the client, `ClientError` can be used for that.

Reply:
Yep, thank you - fixed to throw a clientError with status 400.